ext/oci8/oci8_interface.c

```diff
@@ -608,30 +570,23 @@ PHP_FUNCTION(oci_lob_size)
    Writes data to current position of a LOB */
 PHP_FUNCTION(oci_lob_write)
 {
-	zval *tmp, *z_descriptor = getThis();
+	zval *tmp, *z_descriptor;
 	php_oci_descriptor *descriptor;
 	size_t data_len;
 	zend_long write_len = 0;
 	ub4 bytes_written;
 	char *data;
 
-	if (getThis()) {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "s|l", &data, &data_len, &write_len) == FAILURE) {
-			RETURN_THROWS();
-		}
+	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "Os|l", &z_descriptor, oci_lob_class_entry_ptr, &data, &data_len, &write_len) == FAILURE) {
+		RETURN_THROWS();
+	}
 
-		if (ZEND_NUM_ARGS() == 2) {
-			data_len = MIN((zend_long) data_len, write_len);
-		}
+	if (getThis() && ZEND_NUM_ARGS() == 2) {
+		data_len = MIN((zend_long) data_len, write_len);
 	}
-	else {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "Os|l", &z_descriptor, oci_lob_class_entry_ptr, &data, &data_len, &write_len) == FAILURE) {
-			RETURN_THROWS();
-		}
 
-		if (ZEND_NUM_ARGS() == 3) {
-			data_len = MIN((zend_long) data_len, write_len);
-		}
+	if (! getThis() && ZEND_NUM_ARGS() == 3) {
```

Comment:
nit: `!getThis()` is the usual style which is used in php-src.

Reply:
Ahh thanks, fixed.